### DISCUSSION TITLE
Generalised HMC with neural networks

### DISCUSSION THREAD

**Discussion by ben18785** [pints-team/pints#1640] [New methods]
[Another](https://arxiv.org/pdf/1711.09268.pdf) paper by Hoffman (inventor of NUTS), that generalises the leapfrog operator that allows them to be scaled dynamically.

**Comment by MichaelClerx** [comment]
It might be worth thinking about at some point whether we can generalise things like the leapfrog method to have some sort of re-usable leapfrog code that different algos can rely on

**Comment by MichaelClerx** [comment]
Then again, it's possible they all take different input arguments, produce slightly different output, and would make reading the code an absolute nightmare :D
